### PR TITLE
KMM-8455 :: Feature Internal :: Split CI pipelines job between macos agent and ubuntu agent

### PR DESCRIPTION
**Pull request by migueltorcha**
**Merge / Pull request information:**

* Asana task link: https://app.asana.com/0/1109863238977521/1203027755108455/f

### CODE REVIEW

**Comment by migueltorcha** [comment]
Guys, note that I have added the gradle plugin that I used to know how to split the executions. It is not needed at all, but it is a nice to have feature. In any case, I can remove it

**Review by joselufo — APPROVED** [review]

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -42,7 +44,58 @@ jobs:
...
+
+      - name: Build
+        run: >
+          ./gradlew 
+          :harmony-kotlin:compileKotlinIosArm64
+          :harmony-kotlin:compileKotlinIosSimulatorArm64
+          :harmony-kotlin:compileKotlinIosX64
+          :harmony-kotlin:iosArm64MetadataJar
+          :harmony-kotlin:iosSimulatorArm64MetadataJar
+          :harmony-kotlin:iosX64MetadataJar
```
Comment (franmontiel) [inline .github/workflows/build.yml:92]: If you just invoke the tests task I am pretty sure that those ones under "Build" are also executed too. Any advantage on declaring them independently?

**Review by franmontiel — COMMENTED** [review]
I have a couple of questions because I am interested in knowing about the changes you introduced from the original script to see if it makes sense to add them in other projects too.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -42,7 +44,58 @@ jobs:
         run: ./gradlew :harmony-kotlin:ktlintCheck :harmony-kotlin:detekt
 
       - name: Build
-        run: ./gradlew :harmony-kotlin:build --stacktrace
+        run: ./gradlew :harmony-kotlin:assemble
+
+      - name: Test
+        run: ./gradlew :harmony-kotlin:check
```
[the remark this answers is not in the document]
Comment (migueltorcha) [reply .github/workflows/build.yml:50]: it is just to have clear where the CI failed, along the compilation or along the testing. From my point of view, it is good to have small steps, so anyone could clearly understand where the problem is. On the other hand, you can also analyze how much time is consuming each step.

File: .github/workflows/build.yml (hunk 1 above)
Comment (migueltorcha) [reply .github/workflows/build.yml:92]: the idea is the same than in the previous comment. I prefer to split the job in small steps.

File: .github/workflows/build.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (franmontiel) [reply .github/workflows/build.yml:50]: Ok, make sense.

**Review by franmontiel — APPROVED** [review]

**Pull request merged**